backend/src/main/java/org/acme/TimetableResource.java

```diff
@@ -62,11 +66,61 @@ public List<Timetable> view() {
         return Timetable.listAll();
     }
 
-    @Path("/unit")
-    @POST
+    /**
+     * Update the time/location of a unit once users make 
+     * drag-and-drop modifications in frontend
+     * 
+     * @param updatedUnit   Unit object with updated time/location
+     */
+    @Path("/update")
+    @PUT
+    @Transactional
     @Consumes(MediaType.APPLICATION_JSON)
-    public Unit handleUnit(Unit unit) {
-        return unit;
+    public Response unitUpdate(Unit updatedUnit) {
+
+        List<Unit> units = Unit.listAll();
+
+        // find existing Unit obj in db by unitId
+        for (Unit unit : units) {
+            if (updatedUnit.unitId == unit.unitId) {
+                assert(unit.isPersistent());
+                // update day of week
+                unit.dayOfWeek = updatedUnit.dayOfWeek;
+                // update startTime
+                unit.startTime = updatedUnit.startTime;
+                // if room is different
+                if (unit.room.roomCode != updatedUnit.room.roomCode 
+                || !unit.room.buildingId.equals(updatedUnit.room.buildingId)) {
+                    // update room
+                    Room newRoom;
+                    if ((newRoom = findRoom(updatedUnit.room)) == null) {
+                        throw new WebApplicationException("Room with building ID " + updatedUnit.room.buildingId + ", and room code " + updatedUnit.room.roomCode + " not found", 404);
+                    }
+                    unit.room = newRoom;
+                }
+                return Response.ok().build();
+            }
+        }
+        return Response.serverError().build();
+    }
+
+    /**
+     * Find existing Room object inside the database with 
+     * desired buildingId and roomCode
+     * 
+     * @param inputRoom     Room object with buildingId and roomCode of desired Room
+     * @return              Room object meeting the input criteria, null otherwise
+     */
+    public Room findRoom(Room inputRoom) {
+        List<Room> rooms = Room.listAll();
+        for (Room room : rooms) {
+            if (room.roomCode.equals(inputRoom.roomCode) 
+            && room.buildingId.equals(inputRoom.buildingId)) {
+                assert(room.isPersistent());
+                return room;
+            }
+        }
+        return null;
     }
```

Comment:
hey Lam could you remove the methods from kan-80 from this current PR? 
this block of code is more relevant to kan-80 and i believe wouldn't be appropriate to appear here